worlds/mm2/__init__.py

```diff
@@ -0,0 +1,312 @@
+        total_checks = 24
+        if self.options.consumables in (self.options.consumables.option_1up_etank,
+                                        self.options.consumables.option_all):
+            total_checks += 20
+        if self.options.consumables in (self.options.consumables.option_weapon_health,
+                                        self.options.consumables.option_all):
+            total_checks += 27
+        remaining = total_checks - len(itempool)
+        itempool.extend([self.create_item(name)
+                         for name in self.multiworld.random.choices(list(filler_item_weights.keys()),
+                                                                    weights=list(filler_item_weights.values()),
+                                                                    k=remaining)])
+        self.multiworld.itempool += itempool
+
+    set_rules = set_rules
+
+    def generate_early(self) -> None:
+        if (not self.options.yoku_jumps
+            and self.options.starting_robot_master.current_key == "heat_man") or \
+                (not self.options.enable_lasers
+                 and self.options.starting_robot_master.current_key == "quick_man"):
+            robot_master_pool = [1, 2, 3, 5, 6, 7, ]
+            if self.options.yoku_jumps:
+                robot_master_pool.append(0)
+            if self.options.enable_lasers:
+                robot_master_pool.append(4)
+            self.options.starting_robot_master.value = self.random.choice(robot_master_pool)
+            logger.warning(
+                f"Incompatible starting Robot Master, changing to "
+                f"{self.options.starting_robot_master.current_key.replace('_', ' ').title()}")
+
+    def generate_basic(self) -> None:
+        goal_location = self.multiworld.get_location(dr_wily, self.player)
+        goal_location.place_locked_item(MM2Item("Victory", ItemClassification.progression, None, self.player))
+        self.multiworld.completion_condition[self.player] = lambda state: state.has("Victory", self.player)
+
+    def fill_hook(self,
+                  progitempool: List["Item"],
+                  usefulitempool: List["Item"],
+                  filleritempool: List["Item"],
+                  fill_locations: List["Location"]) -> None:
+        # on a solo gen, fill can try to force Wily into sphere 2, but for most generations this is impossible
+        # since MM2 can have a 2 item sphere 1, and 3 items are required for Wily
+        if self.multiworld.players > 1:
+            return  # Don't need to change anything on a multi gen, fill should be able to solve it with a 4 sphere 1
+        rbm_to_item = {
+            0: heat_man_stage,
+            1: air_man_stage,
+            2: wood_man_stage,
+            3: bubble_man_stage,
+            4: quick_man_stage,
+            5: flash_man_stage,
+            6: metal_man_stage,
+            7: crash_man_stage
+        }
```

Comment:
Any specific reason this isn't a list?

Reply:
I feel like a dict better illustrates the relationship between them, rather than "This is just how they're ordered".

Reply:
Javascript moment